packages/bpk-component-navigation-bar/src/BpkNavigationIconButton.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import { cssModules } from 'bpk-react-utils';
+import BpkButton from 'bpk-component-button';
```

Comment:
Im not sure a `BpkButton` should be used here. Technically the configuration you are using it in is unsupported - hence why you have to add a class to it to remove it's padding. 

If consumers did this we'd say that they were violating the internals of the component therefore we shouldn't be doing it either. If we find ourselves doing this then we should ask why and if it's required it should be made an explicit api (i.e. through a react prop.)

Regardless, I think what you really want to use is the `bpk-component-close-button` component - see this comment on the intention with future direction: https://github.com/Skyscanner/backpack/pull/644#discussion_r181335150

Reply:
The problem with the bpk close button is that we cannot customise the icon, witch is the intention of this component. As for the padding, is being removed because that's how they should look in the nav bar. 

If I were to not use the `BpkButton` component I'd have to implement something similar to the close button that accepts different icons ( which is fine). I just didn't see any reason not to use `BpkButton` and change a bit it's style considering it accepts a `className` prop to do so.